### PR TITLE
[PREVIEW] SIDM-938 prepopulate selfRegister fields

### PR DESCRIPTION
**Pull request by nikola-naydenov-hmcts**
Allow service teams to pre-populate the self register form

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java [hunk 1]
```diff
@@ -162,6 +169,21 @@ public String selfRegister(
         return PAGE_NOT_FOUND_VIEW;
     }
 
+    private void parseFormData(String formData, Model model) {
+        if (formData != null) {
+            try {
+                final RegisterFormData data = mapper
+                    .readerFor(RegisterFormData.class)
+                    .readValue(Base64.decode(formData.getBytes()));
```
Comment (dfourn) [inline src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java]: Is this safe? As in, would it be possible for a potential attack to inject malicious code?

File: src/test/java/uk/gov/hmcts/reform/idam/web/UserControllerTest.java [hunk 2]
```diff
@@ -13,54 +13,7 @@
...
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.USER_ACTIVATION_TOKEN;
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.USER_ACTIVATION_VIEW_NAME;
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.USER_CREATED_VIEW_NAME;
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.USER_EMAIL;
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.USER_EMAIL_PARAMETER;
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.USER_FIRST_NAME;
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.USER_LAST_NAME;
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.USER_PASSWORD;
-import static uk.gov.hmcts.reform.idam.web.util.TestConstants.VALIDATE_TOKEN_ENDPOINT;
+import static uk.gov.hmcts.reform.idam.web.util.TestConstants.*;
```
Comment (dfourn) [inline src/test/java/uk/gov/hmcts/reform/idam/web/UserControllerTest.java]: 👍

File: src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java (hunk 1 above)
Comment (tbamido) [reply src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java]: good point, I think we need to change how we render it in the jsp to disable any scripts that could potentially be injected

Comment (dfourn) [reply src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java]: Try injecting this string into this method and look at the console output: 
`eyJmaXJzdE5hbWUiOiJKb2huIiwibGFzdE5hbWUiOiJEb2UiLCJlbWFpbCI6ImpvaG4uZG9lQGVtYWlsLmNvbSJ9`

Comment (dfourn) [reply src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java]: https://www.nccgroup.trust/globalassets/our-research/us/whitepapers/2018/jackson_deserialization.pdf

Comment (nikola-naydenov-hmcts) [reply src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java]: I have added owasp html encoding

File: src/test/java/uk/gov/hmcts/reform/idam/web/UserControllerTest.java (hunk 2 above)
Comment (tbamido) [reply src/test/java/uk/gov/hmcts/reform/idam/web/UserControllerTest.java]: 👍 nice. we need to come up with a hard number, 10? 20?

**Pull request closed without merging**